### ISSUE TITLE
Bug: `bbox_to_mask3d` fills the whole volume when a box spans or overhangs a full axis

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4255]
`bbox_to_mask3d` builds its mask as a union of three axis slabs and then recovers the intersection with `all` reductions. When a box covers an entire output axis, the union intermediate is `True` everywhere, every reduction comes back `True`, and the function fills the whole volume, losing the bounds on the other two axes. A box that overhangs the volume on one axis, which is the normal state of a box after a crop or a translation, triggers the same failure. `Boxes3D.to_mask` on the same box fills the intended region on both its loop and export paths.

Reproduced 2026-09-05 at `1eea5835`, PyTorch 2.14.0, CPU.

## Reproduction

```python
import torch
from kornia.geometry.bbox import bbox_to_mask3d
from kornia.geometry.boxes import Boxes3D

size = (4, 4, 5)  # depth, height, width
for label, xyzxyz in {
    "x spans 0..4 (full width)": [0., 1., 1., 4., 2., 2.],
    "y spans 0..3 (full height)": [1., 0., 1., 2., 3., 2.],
    "z spans 0..3 (full depth)": [1., 1., 0., 2., 2., 3.],
    "x overhangs -1..5": [-1., 1., 1., 5., 2., 2.],
    "interior 1..2": [1., 1., 1., 2., 2., 2.],
}.items():
    b = Boxes3D.from_tensor(torch.tensor([xyzxyz]), "xyzxyz_plus")
    print(label, int(bbox_to_mask3d(b.data, size).sum()), int(b.to_mask(*size).sum()))
```

## Observed

```text
x spans 0..4 (full width)  80 20
y spans 0..3 (full height) 80 16
z spans 0..3 (full depth)  80 16
x overhangs -1..5          80 20
interior 1..2              8 8
```

The volume has 80 voxels. Every full-axis case fills all of them; only the interior box matches the method.

## Expected

The intersection of the three slabs, as `Boxes3D.to_mask` computes it: 20, 16, 16, 20 and 8 voxels. The three-way `|` followed by `all` reductions should be a three-way `&` of the broadcast slabs, which is what the export path of `Boxes3D.to_mask` already does.

## Cause

In `kornia/geometry/bbox.py`, `bbox_to_mask3d` computes `m = z_slab | y_slab | x_slab` and then `cond1 = m.all(dim=3).all(dim=2)` and its two siblings. For a slab `X` that covers every index, `all_{z,y}(Z | Y | X) = X | Zfull | Yfull` is `True` for every `x`, and the same happens to the other two conditions, so the product is `True` on the whole grid. For interior boxes the `all` reductions recover the slabs exactly, which is why the existing interior-box tests pass.

## Classification

Broken output. A wrong mask for a valid, in-range box. Not a rounding or convention question: the inclusive-truncating policy (#4015) and the `float32` channel-axis output (#4250) are separate.

## Disposition

**Fix directly, not in the window.** The canonical result is unambiguous and `Boxes3D.to_mask`'s export path already implements it; the repair changes no output for any box that the current code handles correctly. The conventions batch 4c PR (#4253) documents the current behavior in the `bbox_to_mask3d` warning and pins it: `test_wart_bbox_to_mask3d_fills_the_whole_volume_when_a_box_spans_an_axis_<this issue>` records today's output, and a strict `xfail` on the intersection result turns into an XPASS when this is fixed.

Found by a review of #4253 during the bbox/boxes conventions audit (#4020); the audit changes no runtime behavior.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
